src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
+const healPoints = require('./healPoints');
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = healPoints(100);
+    Animal.alive.push(this);
+  }
+
+  static removeDiedAmimals() {
+    Animal.alive = Animal.alive.filter(aliveAnimal => aliveAnimal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, damage = 50) {
+    super(name);
+    this.damage = damage;
+  }
+
+  bite(animal) {
+    animal instanceof Herbivore && !animal.hidden
+      ? animal.health -= this.damage
+      : animal.health = this.health;
```

Comment:
Why you use ternary operator here?
We use ternary operator only if we do smth in every case, but here you do smth only if condition `true`.

```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
      animal.health -= this.damage
    }
```